src/vm/analysis/type_checker/mod.rs

```diff
@@ -201,12 +204,22 @@ impl <'a, 'b> TypeChecker <'a, 'b> {
                 self.type_check(&exp, &local_context)?;
             }
         }
-
         Ok(())
     }
 
     // Type check an expression, with an expected_type that should _admit_ the expression.
     pub fn type_check_expects(&mut self, expr: &SymbolicExpression, context: &TypingContext, expected_type: &TypeSignature) -> TypeResult {
+        
+        match (&expr.expr, expected_type) {
+            (LiteralValue(Value::Principal(PrincipalData::Contract(ref contract_identifier))), TypeSignature::TraitReferenceType(trait_identifier)) => {
+                let traits = self.db.get_implemented_traits(&contract_identifier).unwrap();
```

Comment:
This check will fail _accidental_ implementors of the trait, which I think we should support even in this case.

Reply:
Oh ok, that was supposed to be a feature, not a bug :).
Will revisit.

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1246/commits/058549a72fd3e882bcef0b45ccd31c12a55051ee